src/python/pants/backend/python/rules/python_test_debugger.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import Path
+
+from pants.backend.python.rules.inject_init import InjectedInitDigest
+from pants.backend.python.rules.pex import (
+  CreatePex,
+  Pex,
+  PexInterpreterConstraints,
+  PexRequirements,
+)
+from pants.backend.python.subsystems.pytest import PyTest
+from pants.backend.python.subsystems.python_setup import PythonSetup
+from pants.base.build_root import BuildRoot
+from pants.build_graph.address import Address
+from pants.engine.fs import Digest, DirectoriesToMerge, DirectoryToMaterialize, Workspace
+from pants.engine.interactive_runner import InteractiveProcessRequest, InteractiveRunner
+from pants.engine.legacy.graph import BuildFileAddresses, HydratedTarget, TransitiveHydratedTargets
+from pants.engine.legacy.structs import PythonTestsAdaptor
+from pants.engine.rules import rule
+from pants.engine.selectors import Get, MultiGet
+from pants.rules.core.core_test_model import TestDebugResult
+from pants.rules.core.strip_source_root import SourceRootStrippedSources
+from pants.util.contextutil import temporary_dir
+
+
+@rule(name="Run pytest in an interactive process")
+async def debug_python_test(
+  test_target: PythonTestsAdaptor,
+  pytest: PyTest,
+  python_setup: PythonSetup,
+  build_root: BuildRoot,
+  runner: InteractiveRunner,
+  workspace: Workspace,
```

Comment:
Are we still conceptualizing `InteractiveRunner` and `Workspace` as types that should only be requestable from a `@console_rule`? @stuhood